src/AI/AIUtils.java

```diff
@@ -206,7 +206,7 @@ public static GameAction moveTowardLocation(Unit unit, XYCoord destination, Game
 
     if( path.getPathLength() > 0 && validMoves.size() > 0 ) // Check that the destination is reachable at least in theory.
     {
-      path.snip(unit.model.movePower+1); // Trim the path so we go the right immediate direction.
+      path.snip(unit.getMovePower(gameMap)+1); // Trim the path so we go the right immediate direction.
```

Comment:
Should we change the name of `model.movePower` to `baseMovePower` so it's clear that it's no longer the final word?

I guess we could also make it inaccessible, but I'm less a fan of that.

Reply:
I suppose.
Doing the code change brought to mind our current design of building the BattleMech unit type with Tech. I can see three options here, for when we move to having immutable `UnitModel`s:
1) Don't actually make their fields `final`, so she can just make a copy, modify it, and put it in the global unit type list.
2) Do make their fields `final`, and provide a function to make a modified copy so she can do the above.
3) Force all `UnitModelScheme`s to define a BattleMech unit type.

I think we were planning to do 3 before, but we do have the option of 1 and 2. Thoughts?

Reply:
I'm fine with sticking to 3, but I'm not necessarily opposed to the others either.

Reply:
I think I'll stick to 1 for the immediate future, in the spirit of KISS.